src/Illuminate/Events/EventHooks.php

```
@@ -0,0 +1,615 @@
+     *
+     * @param  array<int, mixed>  $payload
+     */
+    protected function hasHierarchicalCallbacks(string $hook, string $event, array $payload): bool
+    {
+        $this->validateHook($hook);
+
+        if (! class_exists($event) && ! is_object(Arr::first($payload))) {
+            return false;
+        }
+
+        if (! blank($this->cache['has_hierarchical'][$event]) || ! blank($this->cache['has_hierarchical'][get_class(Arr::first($payload))])) {
```

PHPMD (2.15.0)

Avoid using static access to class '\Illuminate\Support\Arr' in method 'hasHierarchicalCallbacks'. (Clean Code Rules)

(StaticAccess)